src/sparse/primitive.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{
+    common::{error::DeserializeError, Bytes32, Prefix, PrefixError},
+    sparse::Node,
+};
+
+/// **Leaf buffer:**
+///
+/// | Allocation | Data                       |
+/// |------------|----------------------------|
+/// | `00 - 04`  | Height (4 bytes)           |
+/// | `04 - 05`  | Prefix (1 byte, `0x00`)    |
+/// | `05 - 37`  | hash(Key) (32 bytes)       |
+/// | `37 - 69`  | hash(Data) (32 bytes)      |
+///
+/// **Node buffer:**
+///
+/// | Allocation | Data                       |
+/// |------------|----------------------------|
+/// | `00 - 04`  | Height (4 bytes)           |
+/// | `04 - 05`  | Prefix (1 byte, `0x01`)    |
+/// | `05 - 37`  | Left child key (32 bytes)  |
+/// | `37 - 69`  | Right child key (32 bytes) |
+///
+pub type Primitive = (u32, u8, Bytes32, Bytes32);
+
+trait PrimitiveView {
```

Comment:
Hmmm, if we still implement a getter for each field, maybe it is better to represent `(u32, u8, Bytes32, Bytes32)` as `[u8; 69]`. In this case,  we don't have any overhead regarding serialization/deserialization and it is simple `mem::copy`

Reply:
Also, you can implement the `PrimitiveView` trait for `& Primitive`, and in this case you can implement `TryFrom<&Primitive>` -> you don't need `into_owned`

Reply:
I see that you tried to move from `Buffer = [u8; 69]` to a more user-friendly type. But you don't use tuple syntax like `Primitive.0` because of `PrimitiveView`, so we still can have the buffer under the hood=)

Reply:
the byte buffer approach still requires similar overhead to serialization unless unsafe zero-copy methods are used (which this PR originally did). However, this tuple was a way to avoid needing to derive seralization traits or use unsafe as an alternative.

cc @freesig

Reply:
We need to do less copy=) [Here](https://github.com/FuelLabs/fuel-merkle/pull/138) we don't need to use `into_owned`